### ISSUE TITLE
Clarifications needed

### ISSUE THREAD

**Issue by ChubaOraka** [LinkedInLearning/pytorch-essential-training-working-with-images-3968391#1]
I was looking for where the output layers were replaced. I could in the code. The only place where this is explicitly stated is this line on [branch `04_07`](https://github.com/LinkedInLearning/pytorch-essential-training-working-with-images-3968391/blob/04_07/04_07.ipynb):

```Python
## change to output 2 logits
model_ft.fc = nn.Linear(num_ftrs, 1)
```

It seems `BCEWithLogitsLoss` loss function (used in [branch `04_07`](https://github.com/LinkedInLearning/pytorch-essential-training-working-with-images-3968391/blob/04_07/04_07.ipynb)) needs this but `CrossEntropyLoss` loss function (used in [`04_05`](https://github.com/LinkedInLearning/pytorch-essential-training-working-with-images-3968391/blob/04_05/04_05.ipynb) and earlier branches) doesn't. So I would like to know how `CrossEntropyLoss` identifies the needed number of nodes in the output layer. Does it get the needed number of nodes from the `model_ft.parameters` argument of the optimizer function? :

```Python
optimizer_ft = optim.SGD(model_ft.parameters(), lr=0.0001, momentum=0.9)
```

Secondly, the course refers to freezing the non-output layers during training. It doesn't seem like this was done in the code.

Finally, the course suggests we use [Google Colab](https://colab.research.google.com/), but the [repo](https://github.com/LinkedInLearning/pytorch-essential-training-working-with-images-3968391/tree/main) includes configurations for [`devcontainer`](https://github.com/LinkedInLearning/pytorch-essential-training-working-with-images-3968391/blob/main/.devcontainer/devcontainer.json). Can you include instructions on how to set up the project with `devcontainer` or can it also be used on Google Colab?
